src/tools/rados/rados.cc

```diff
@@ -460,6 +461,53 @@ static int do_put(IoCtx& io_ctx, RadosStriper& striper,
   return ret;
 }
 
+static int do_append(IoCtx& io_ctx, RadosStriper& striper,
+                  const char *objname, const char *infile, int op_size,
+                  bool use_striper)
+{
+  string oid(objname);
+  bufferlist indata;
+  bool stdio = false;
+  if (strcmp(infile, "-") == 0)
+    stdio = true;
+
+  int ret;
+  int fd = STDIN_FILENO;
+  if (!stdio)
+    fd = open(infile, O_RDONLY);
+  if (fd < 0) {
+    cerr << "error reading input file " << infile << ": " << cpp_strerror(errno) << std::endl;
+    return 1;
+  }
+  char *buf = new char[op_size];
+  int count = op_size;
+  while (count != 0) {
+    count = read(fd, buf, op_size);
+    if (count < 0) {
+      ret = -errno;
+      cerr << "error reading input file " << infile << ": " << cpp_strerror(ret) << std::endl;
+      goto out;
+    }
+    indata.append(buffer::ptr(buffer::create_static(count, buf)));
+    if (use_striper) {
+      ret = striper.append(oid, indata, count);
+    } else {
+      ret = io_ctx.append(oid, indata, count);
+    }
+    indata.clear();
+
+    if (ret < 0) {
```

Comment:
i am not sure we need to do the error handling this way. `append()` returns `0` on success, and a negative number on failure. so we can just let `while (count != 0)` take care of the loop condition, and use `break` if ret is not `0`.